src/erlcloud_iam.erl

```diff
@@ -707,26 +709,54 @@ get_credential_report(Config) ->
     iam_query(Config, "GetCredentialReport", [], ItemPath, DataTypeDef).
 
 simulate_principal_policy(PolicySourceArn, ActionNames) ->
-    simulate_principal_policy(PolicySourceArn, ActionNames, default_config()).
+    simulate_principal_policy(PolicySourceArn, ActionNames, [], [], default_config()).
 
-simulate_principal_policy(PolicySourceArn, ActionNames, #aws_config{} = Config)
-  when is_list(PolicySourceArn) andalso is_list(ActionNames) ->
+simulate_principal_policy(PolicySourceArn, ActionNames, #aws_config{} = Config) ->
+    simulate_principal_policy(PolicySourceArn, ActionNames, [], [], Config);
+
+simulate_principal_policy(PolicySourceArn, ActionNames, ResourceArns) ->
+    simulate_principal_policy(PolicySourceArn, ActionNames, ResourceArns, [], default_config()).
+
+simulate_principal_policy(PolicySourceArn, ActionNames, ResourceArns, #aws_config{} = Config) ->
+    simulate_principal_policy(PolicySourceArn, ActionNames, ResourceArns, [], Config);
+
+simulate_principal_policy(PolicySourceArn, ActionNames, ResourceArns, ContextEntries) ->
+    simulate_principal_policy(PolicySourceArn, ActionNames, ResourceArns, ContextEntries, default_config()).
+
+simulate_principal_policy(PolicySourceArn, ActionNames, ResourceArns, ContextEntries, #aws_config{} = Config)
```

Comment:
should we check that list of action is not empty? I think it will be valuable...